File: springboot-jwt/src/main/java/com/grepp/spring/infra/response/WebSocketResponse.java
package com.grepp.spring.infra.response;

public record WebSocketResponse<T>(
    String code,
    String message,
    T data
) {
    
    public static <T> WebSocketResponse<T> success(T data) {
        return new WebSocketResponse<>(ResponseCode.OK.code(), ResponseCode.OK.message(), data);
    }
    
    public static <T> WebSocketResponse<T> noContent() {
        return new WebSocketResponse<>(ResponseCode.OK.code(), ResponseCode.OK.message(), null);
    }
    
    public static <T> WebSocketResponse<T> error(ResponseCode code) {
        return new WebSocketResponse<>(code.code(), code.message(), null);
    }
    
    public static <T> WebSocketResponse<T> error(ResponseCode code, T data) {
        return new WebSocketResponse<>(code.code(), code.message(), data);
    }
}

File: chat-service-A/src/main/kotlin/com/grepp/spring/infra/response/WebSocketResponse.kt
package com.grepp.spring.infra.response

@JvmRecord
data class WebSocketResponse<T>(
    val code: String,
    val message: String,
    val data: T
) {
    companion object {
        fun <T> success(data: T): WebSocketResponse<T> {
            return WebSocketResponse<T>(ResponseCode.OK.code(), ResponseCode.OK.message(), data)
        }

        fun <T> noContent(): WebSocketResponse<T?> {
            return WebSocketResponse<T?>(ResponseCode.OK.code(), ResponseCode.OK.message(), null)
        }

        fun <T> error(code: ResponseCode): WebSocketResponse<T?> {
            return WebSocketResponse(code.code(), code.message(), null)
        }

        fun <T> error(code: ResponseCode, data: T): WebSocketResponse<T> {
            return WebSocketResponse(code.code(), code.message(), data)
        }
    }
}
